cfme/tests/cloud/test_cloud_timelines.py

```diff
@@ -1,147 +1,311 @@
+
+        for evt in events_list:
+            try:
+                if not policy_events:
+                    if evt.source_instance in self.inst.name and evt.event_type in self.tl_event:
+                        found_events.append(evt)
+                        break
+                else:
+                    if evt.event_type in self.tl_event and evt.target in self.inst.name:
+                        found_events.append(evt)
+                        break
+            except AttributeError as err:
+                logger.warn('Issue with TimelinesEvent: %r .Faulty event: %r', err, evt)
+                continue
+
+        logger.info('found events on {tgt}: {evt}'.format(
+            tgt=target, evt="\n".join([repr(e) for e in found_events])))
+
+        return len(found_events)
+
+    def catch_in_timelines(self, soft_assert, targets, policy_events=False):
+        for target in targets:
+            try:
+                wait_for(self._check_timelines,
+                         [target, policy_events],
+                         timeout='7m',
+                         fail_condition=0)
+            except TimedOutError:
+                soft_assert(False, '0 occurrence of {} found on the timeline of {}'.format(
+                    self.event, target))
+
+
+def test_cloud_timeline_create_event(new_instance, soft_assert, azone):
+    targets = (new_instance, new_instance.provider, azone)
+    event = 'create'
+    inst_event = InstEvent(new_instance, event)
+    logger.info('Will generate event %r on machine %r', event, new_instance.name)
+    wait_for(inst_event.emit, timeout='9m', message='Event {} did timeout'.format(event))
+    inst_event.catch_in_timelines(soft_assert, targets)
+
+
+@pytest.mark.meta(blockers=[BZ(1542962, forced_streams=['5.8'])])
+def test_cloud_timeline_policy_event(new_instance, control_policy, soft_assert):
+    event = 'policy'
+    targets = (new_instance, new_instance.provider)
+    inst_event = InstEvent(new_instance, event)
+    wait_for(inst_event.emit, timeout='9m', message='Event {} did timeout'.format(event))
+    inst_event.catch_in_timelines(soft_assert, targets, policy_events=True)
 
 
-def test_cloud_instance_event(gen_events, new_instance):
-    """ Tests vm events on timelines
+@pytest.mark.meta(blockers=[BZ(1534400, unblock=lambda provider: not provider.one_of(
+    AzureProvider))])
+def test_cloud_timeline_diagnostic(new_instance, mark_vm_as_appliance, soft_assert):
+    """Check Configuration/diagnostic/timelines. """
+    event = 'start'
+    targets = (new_instance.appliance.server,)
+    inst_event = InstEvent(new_instance, event)
+    inst_event.catch_in_timelines(soft_assert, targets)
 
-    Metadata:
-        test_flag: timelines, provision
-    """
-    wait_for(count_events, [new_instance, new_instance], timeout='5m', fail_condition=0,
-             message="events to appear")
 
+def test_cloud_timeline_power_events(new_instance, soft_assert, azone):
+    events_list = ['stop', 'start']
```

Comment:
**QUESTION**
Can this test be parametrized?

Reply:
i ll divide the test in more part due to the pytest bug